flink-runtime/src/main/java/org/apache/flink/runtime/dispatcher/Dispatcher.java

```diff
@@ -350,71 +341,42 @@ private boolean isPartialResourceConfigured(JobGraph jobGraph) {
 		}, getRpcService().getExecutor());
 	}
 
-	private CompletableFuture<Void> persistAndRunJob(JobGraph jobGraph) throws Exception {
+	private Void persistAndRunJob(JobGraph jobGraph) throws Exception {
```

Comment:
I have this ugly `Void` and `return null` here so that I can call the method as a method reference. 
My next push will contain an attempt to make this cleaner (by introducing a `VoidFunctionWithException` interface)

Reply:
shouldn't this also work if the method has the signature: `void persistAndRunJob(...)`?

Reply:
Unfortunately not: https://stackoverflow.com/questions/28762715/cannot-convert-void-to-java-lang-void

Reply:
Hmm, I think the problem is that you are using `persistAndRunJob` still as a function even though it does not return anything. Changing `waitForTerminatingJobManager` to take a `ThrowingConsumer` should solve the problem.

Reply:
Ah, very nice. Thanks a lot!